### PR TITLE
AIs trade techs. Also modified AI tech choose strategy.

### PR DESCRIPTION
**Pull request by ninjatao**
https://github.com/yairm210/UnCiv/issues/416

Code (almost) ready. Need some test to see if it is overpowered.

It turns out ai is exchanging techs TOO fast :P    Need some improvement about this.

### CODE REVIEW

File: core/src/com/unciv/logic/automation/NextTurnAutomation.kt [hunk 1]
```diff
@@ -22,11 +25,43 @@ class NextTurnAutomation{
...
         if (civInfo.tech.techsToResearch.isEmpty()) {
-            val researchableTechs = GameBasics.Technologies.values.filter { civInfo.tech.canBeResearched(it.name) }
-            val techToResearch = researchableTechs.groupBy { it.cost }.minBy { it.key }!!.value.getRandom()
-            civInfo.tech.techsResearched.add(techToResearch.name)
+            val researchableTechs = GameBasics.Technologies.values.filter { !civInfo.tech.isResearched(it.name) && civInfo.tech.canBeResearched(it.name) }
+            val techsGroups = researchableTechs.groupBy { it.cost }
+            val costs = techsGroups.keys.sorted()
+
+            val tech: Technology
+            if (techsGroups[costs[0]]!!.size == 1 || costs.size == 1) {
```
Comment (yairm210) [inline core/src/com/unciv/logic/automation/NextTurnAutomation.kt]: Techgroups[costs[0]] should probably be put in a separate parameter

Comment (ninjatao) [reply core/src/com/unciv/logic/automation/NextTurnAutomation.kt]: Like how? I don't quite get what you mean...

Comment (yairm210) [reply core/src/com/unciv/logic/automation/NextTurnAutomation.kt]: Var cheapestTechs = Techgroups[costs[0]]

For readability

**Comment by ninjatao** [comment]
Got conflict with other commits. Closing this pr and refreshing this branch.

**Pull request closed without merging**